disperser/common/blobstore/blob_metadata_store.go

```diff
@@ -203,6 +206,73 @@ func (s *BlobMetadataStore) GetAllBlobMetadataByBatch(ctx context.Context, batch
 	return metadatas, nil
 }
 
+// GetBlobMetadataByStatusWithPagination returns all the metadata with the given status upto the specified limit
+// along with items, also returns a pagination token that can be used to fetch the next set of items
+//
+// Note that this may not return all the metadata for the batch if dynamodb query limit is reached.
+// e.g 1mb limit for a single query
+func (s *BlobMetadataStore) GetAllBlobMetadataByBatchWithPagination(
+	ctx context.Context,
+	batchHeaderHash [32]byte,
+	limit int32,
+	exclusiveStartKey *disperser.BatchIndexExclusiveStartKey,
+) ([]*disperser.BlobMetadata, *disperser.BatchIndexExclusiveStartKey, error) {
+	var attributeMap map[string]types.AttributeValue
+	var err error
+
+	// Convert the exclusive start key to a map of AttributeValue
+	if exclusiveStartKey != nil {
+		attributeMap, err = convertToAttribMapBatchIndex(exclusiveStartKey)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	queryResult, err := s.dynamoDBClient.QueryIndexWithPagination(
```

Comment:
Does this return error if 1mb limit is reached?

Reply:
No, it simply just returns less items than the limit.

> A single Query operation will read up to the maximum number of items set (if using the Limit parameter) or a maximum of 1 MB of data